.github/workflows/build_and_test.yml

```
@@ -882,6 +857,28 @@ jobs:
                                                369495373322.dkr.ecr.eu-central-1.amazonaws.com/${image}:${{ needs.tag.outputs.build-tag }}
           done
 
+  push-to-acr-dev:
+    if: github.ref_name == 'main'
+    needs: [ tag, promote-images ]
+    uses: ./.github/workflows/_push-to-acr.yml
+    with:
+      client_id: ${{ vars.AZURE_DEV_CLIENT_ID }}
+      image_tag: ${{ needs.tag.outputs.build-tag }}
+      registry_name: ${{ vars.AZURE_DEV_REGISTRY_NAME }}
+      tenant_id: ${{ vars.AZURE_TENANT_ID }}
+      subscription_id: ${{ vars.AZURE_DEV_SUBSCRIPTION_ID }}
+
+  push-to-acr-prod:
+    if: github.ref_name == 'main'
+    needs: [ tag, push-to-acr-dev ]
```

I'd keep it like this to have less changes in the future

```suggestion
    needs: [ tag, promote-images ]
```